iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/digitaltwin/DigitalTwinClientComponentTests.java

```diff
@@ -0,0 +1,211 @@
+        String componentName = "thermostat1";
+        String commandName = "getMaxMinReport";
+        String commandInput = ZonedDateTime.now(ZoneOffset.UTC).minusMinutes(5).format(DateTimeFormatter.ISO_DATE_TIME);
+        DigitalTwinInvokeCommandRequestOptions options = new DigitalTwinInvokeCommandRequestOptions();
+        options.setConnectTimeoutInSeconds(15);
+        options.setResponseTimeoutInSeconds(15);
+
+        // setup device callback
+        Integer deviceSuccessResponseStatus = 200;
+        String deviceSuccessResponseMessage = "Success";
+        Integer deviceFailureResponseStatus = 500;
+        String deviceFailureResponseMessage = "Failed";
+
+        // Device method callback
+        String componentCommandName = componentName + "*" + commandName;
+        DeviceMethodCallback deviceMethodCallback = (methodName, methodData, context) -> {
+            if(methodName.equalsIgnoreCase(componentCommandName)) {
+                return new DeviceMethodData(deviceSuccessResponseStatus, deviceSuccessResponseMessage);
+            }
+            else {
+                return new DeviceMethodData(deviceFailureResponseStatus, deviceFailureResponseMessage);
+            }
+        };
+
+        // IotHub event callback
+        IotHubEventCallback iotHubEventCallback = (responseStatus, callbackContext) -> {};
+
+        deviceClient.subscribeToDeviceMethod(deviceMethodCallback, commandName, iotHubEventCallback, commandName);
+
+        // act
+        DigitalTwinCommandResponse responseWithNoPayload = this.digitalTwinClient.invokeComponentCommand(deviceId, componentName, commandName, null);
+        DigitalTwinCommandResponse response = this.digitalTwinClient.invokeComponentCommand(deviceId, componentName, commandName, commandInput);
+        ServiceResponseWithHeaders<DigitalTwinCommandResponse, DigitalTwinInvokeCommandHeaders> responseWithHeaders = this.digitalTwinClient.invokeComponentCommandWithResponse(deviceId, componentName, commandName, commandInput, options);
+
+        // assert
+        String receivedDeviceResponseStatus = "\"" + deviceSuccessResponseMessage + "\"";
+        assertEquals(deviceSuccessResponseStatus, responseWithNoPayload.getStatus());
+        assertEquals(receivedDeviceResponseStatus, responseWithNoPayload.getPayload());
+        assertEquals(deviceSuccessResponseStatus, responseWithNoPayload.getStatus());
+        assertEquals(receivedDeviceResponseStatus, response.getPayload());
+        assertEquals(deviceSuccessResponseStatus, responseWithHeaders.body().getStatus());
+        assertEquals(receivedDeviceResponseStatus, responseWithHeaders.body().getPayload());
+    }
+
+    @Test
+    public void invokeRootLevelCommand() throws IOException {
+        // arrange
+        String commandName = "reboot";
+        String commandInput = "5";
+        DigitalTwinInvokeCommandRequestOptions options = new DigitalTwinInvokeCommandRequestOptions();
+        options.setConnectTimeoutInSeconds(15);
```

Comment:
extract 15 to constant since it's used in multiple tests

Reply:
I felt it would be more readable for a test to have it right there. Specially you might want different tests to do different things. But I'm open to a discussion. For product code, I agree constants are good.